CRM/Civicase/Service/CaseMenu/AllRoute.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * Class for All route.
+ *
+ * Used for listing all the Cases for the associated Category Type Category.
+ */
+class CRM_Civicase_Service_CaseMenu_AllRoute extends CRM_Civicase_Service_CaseMenu_Route {
+
+  /**
+   * {@inheritDoc}
+   */
+  protected function label() {
+    return ts("All %1", [1 => ucfirst(strtolower($this->categoryName))]);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  protected function name() {
+    return "all_{$this->categoryName}";
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  protected function url() {
+    $url = '';
+    $url .= 'civicrm/case/a/';
```

Comment:
These 2 lines can be merged into 1.

Reply:
I do intentionally this, I mean, I initialize the variable and later process the appends.
This prevents errors if you want to switch a given part to another block of the code.
Anyway, if you still oppose the idea, let me know and I will change the occurrences.